app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,46 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def new
+  end
+
+  def create
+    merchant = Merchant.new({
+      name: params[:name],
+      address: params[:address],
+      city: params[:city],
+      state: params[:state],
+      zip: params[:zip],
+      })
+      merchant.save
+      redirect_to '/merchants'
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def edit
+    show
+    # @merchant = Merchant.find(params[:id])
+  end
```

Comment:
In edit, we need to save the merchant we plan on editing into an instance variable so that our form can have it to display any info we want about that merchant, whether it's in a title, or as a placeholder in the text fields. In your routes, it looks like you'll have the id coming in with the path, so what you have commented out should work. I know you can call the show method and get what you want since they are gathering the same thing, but get into the habit of creating each actions own business logic instead of calling on another method.